src/app.js

```diff
@@ -1,23 +1,154 @@
-// starting point of the application
-
 const express = require("express");
+const { connectDB } = require("./config/database");
+const User = require("./models/user");
+const { validateSignUpData } = require("./utils/validation");
+const bcrypt = require("bcrypt");
+
+const app = express();
+
+app.use(express.json());
+
+// POST api - signup
+app.post("/signup", async (req, res, next) => {
+  try {
+    // validate the data
+    validateSignUpData(req);
+
+    // hash the password
+    const { password } = req.body;
+    const hasedPass = await bcrypt.hash(password, 10);
 
-const app = express(); // instance of this express JS application
+    // Creating the instance of the User model
+    const user = new User({ ...req.body, password: hasedPass });
+    // user.save() will return the promise so we have to use async await
+    await user.save();
+    res.send("user added successfully");
+  } catch (err) {
+    res.status(400).send({ message: err.message });
+  }
+});
+
+app.post("/login", async (req, res, next) => {
+  try {
+    // validate email
+    const { email, password } = req.body;
 
-app.use("/", (req, res) => {
-  res.send("Dashboard");
+    const user = await User.findOne({ email });
+    if (!user) {
+      throw new Error("Invalid Credentials");
+    } else {
+      const isPasswordMatching = await bcrypt.compare(password, user.password);
+      if (!isPasswordMatching) {
+        throw new Error("Invalid Credentials");
+      } else {
+        res.send("Login successfull");
+      }
+    }
+  } catch (err) {
+    res.status(400).send({ message: err.message });
+  }
 });
 
-app.use("/hello", (req, res) => {
-  res.send("Hello world");
+//GET - feed
+app.get("/feed", async (req, res, next) => {
+  try {
+    const users = await User.find({});
+    if (!users.length) {
+      res.status(404).send("No users");
+    } else {
+      res.send(users);
+    }
+  } catch (err) {
+    res.status(400).send({ message: err.message });
+  }
 });
 
-app.use("/test", (req, res) => {
-  // request handler function
-  res.send("hello from the server");
+// GET - user
+app.get("/user", async (req, res, next) => {
+  const userEmail = req.body.email;
+
+  try {
+    const users = await User.find({ email: userEmail });
+    if (!users.length) {
+      res.status(404).send("User not found");
+    } else {
+      res.send(users);
+    }
+  } catch (err) {
+    res.status(400).send({ message: err.message });
+  }
 });
 
-const PORT = 3030;
-app.listen(PORT, () => {
-  console.log("Server is successfully listening on port", PORT);
+// GET - firstUser
+app.get("/firstUser", async (req, res, next) => {
+  try {
+    const user = await User.findOne({ email: req.body.email });
+    if (!user) {
+      res.status(404).send("User not found");
+    } else {
+      res.send(user);
+    }
+  } catch (err) {
+    res.status(400).send({ message: err.message });
+  }
 });
+
+// DELETE
+app.delete("/user", async (req, res, next) => {
```

Comment:
routes should be grouped together in a router and then simply added as a middleware to the app